SearchController

barbarakos opened the pull request:
Added custom search bar view and SearchViewController.
![Simulator Screen Shot - iPhone 14 Plus - 2022-11-04 at 15 44 25](https://user-images.githubusercontent.com/101199634/200016148-1105b1da-5806-4136-ace8-3a6f700eef0e.png)
![Simulator Screen Shot - iPhone 14 Plus - 2022-11-04 at 15 44 29](https://user-images.githubusercontent.com/101199634/200016232-37cfb05a-da87-4bc9-bcad-ec625d074867.png)

barbarakos committed:
SearchController

Added custom search bar view and searchViewController.

barbarakos committed:
Merge branch 'feature/QuestionNetwork' into feature/SearchController

# Conflicts:
#	QuizApp.xcodeproj/project.pbxproj

barbarakos committed:
Merge branch 'feature/leaderboard' into feature/SearchController

dorastipkovic commented on QuizApp/Search/SearchViewController.swift:
```diff
@@ -0,0 +1,258 @@
+    func bindViewModel() {
+        quizViewModel
+            .$quizError
+            .sink { [weak self] quizError in
+                if let error = quizError {
+                    self?.handleNoQuizzesAvailable(error: error)
+                }
+            }
```
```suggestion
            .compactMap { $0 }
            .sink { [weak self] quizError in
                self?.handleNoQuizzesAvailable(error: error)
            }
```

dorastipkovic commented on QuizApp/Search/SearchViewController.swift:
```diff
@@ -0,0 +1,258 @@
+                guard let self = self else { return }
+
+                self.applySnapshot(quizzes: quizzes)
+                if !quizzes.isEmpty {
+                    self.quizErrorView.isHidden = true
+                }
```
```suggestion
                self.quizErrorView.isHidden = !quizzes.isEmpty
```

dorastipkovic commented on QuizApp/Search/SearchViewController.swift:
```diff
@@ -0,0 +1,258 @@
+        searchBar.textField.delegate = self
+
+        collectionView.backgroundColor = .clear
+        collectionView.isScrollEnabled = true
```
I believe scrolling is enabled by default

dorastipkovic commented on QuizApp/Search/SearchViewController.swift:
```diff
@@ -0,0 +1,258 @@
+        gradientLayer.locations = [0, 1]
+
+        searchBar.snp.makeConstraints {
+            $0.top.equalTo(view.safeAreaLayoutGuide).offset(-topOffset)
```
```suggestion
            $0.top.equalTo(view.safeAreaLayoutGuide).inset(topOffset)
```

dorastipkovic commented on QuizApp/Search/SearchViewController.swift:
```diff
@@ -0,0 +1,258 @@
+        var snapshot = Snapshot()
+        CategorySection.allCases.forEach { section in
+            let filteredQuizzes = quizzes.filter { $0.category == section.rawValue.uppercased() }
+            if !filteredQuizzes.isEmpty {
+                snapshot.appendSections([section])
+                snapshot.appendItems(filteredQuizzes, toSection: section)
+            }
```
Change to guard

dorastipkovic commented on QuizApp/Search/SearchViewController.swift:
```diff
@@ -0,0 +1,258 @@
+
+    private lazy var dataSource = makeDataSource()
+    private var cancellables = Set<AnyCancellable>()
+    private var quizzes: [QuizModel]!
```
Use quizzes from ViewModel

dorastipkovic commented on QuizApp/Search/Views/SearchBarView.swift:
```diff
@@ -0,0 +1,73 @@
+
+class SearchBarView: UIView {
+
+    var textField: UITextField!
+    var searchButton: UIButton!
```
private

barbarakos committed:
PR changes

Edited dark navigation bar that shows when scrolling, applied PR requested changes.
